src/FieldGuesser.tsx

```diff
@@ -69,7 +70,7 @@ const renderField = (
       <ReferenceArrayField
         {...(props as ReferenceArrayFieldProps)}
         reference={field.reference.name}>
-        <SingleFieldList>
+        <SingleFieldList {...(props as SingleFieldListProps)}>
```

Comment:
Maybe passing `linkType` explicitly would be better to avoid a props "mix".

Reply:
OK, changed.